edgedb-protocol/src/serialization/decode/raw_composite.rs

```diff
@@ -135,6 +151,12 @@ mod inner {
             Ok(Some(self.read_element(len)?))
         }
 
+        pub fn read_object_element(&mut self) -> Result<Option<&'t [u8]>, DecodeError> {
+            ensure!(self.raw.remaining() >= 4, self.underflow());
```

Comment:
```suggestion
            ensure!(self.raw.remaining() >= 8, self.underflow());
```
I would keep it at eight, to give rust a chance to optimize out the following check. And this is consistent with what we're doing everywhere else.